eng/pipelines/templates/steps/build.yml

```diff
@@ -60,28 +60,20 @@ steps:
       Artifacts: ${{ parameters.Artifacts }}
       ArtifactPath: $(Build.SourcesDirectory)/sdk/${{ parameters.ServiceDirectory }}
 
-  # It's important for performance to pass "sdk" as "sourceFolder" rather than as a prefix in "contents".
-  # The task first enumerates all files under "sourceFolder", then matches them against the "contents" pattern.
-  - task: CopyFiles@2
-    inputs:
-      sourceFolder: sdk
-      contents: |
-        **/$(coalesceResultFilter)/*.tgz
-        **/$(coalesceResultFilter)/browser/*.zip
-      targetFolder: $(Build.ArtifactStagingDirectory)
-      flattenFolders: true
-    displayName: "Copy packages"
-
-  - template: /eng/common/pipelines/templates/steps/publish-artifact.yml
-    parameters:
-      ArtifactPath: '$(Build.ArtifactStagingDirectory)'
-      ArtifactName: 'packages'
-
   - template: ../steps/generate-doc.yml
     parameters:
       ServiceDirectory: ${{parameters.ServiceDirectory}}
 
+  - ${{ each artifact in parameters.Artifacts }}:
```

Comment:
In order to remove a lot of the long list of steps in our pipelines we should not loop over the artifacts this way but instead do it inline in the ps script. See https://github.com/Azure/azure-sdk-for-java/blob/master/eng/pipelines/templates/jobs/ci.yml#L98-L113 as an example.